### ISSUE TITLE
False positive with match_same_arms

### ISSUE THREAD

**Issue by jDomantas** [rust-lang/rust-clippy#2185]
If two arms have same values after inlining constants, clippy will report a lint on those arms. Is this a good idea? In my code I had two constants that were semantically different but happened to have the same value, and the suggested replacement did not really make sense.

Here's an example that triggers the lint:

```rust
const FOO: u32 = 4;
const BAR: u32 = 4;

fn main() {
    match 1 {
        1 => FOO,
        2 => BAR,
        _ => 0,
    };
}
```

**Comment by 8573** [comment]
Has this been fixed in commit ff32d5f7343934f4267af4e270017c716fb9d4ad?
